### ISSUE TITLE
windows build?

### ISSUE THREAD

**Issue by bmcfee** [conda-forge/mpg123-feedstock#28]
### Comment:

This package currently skips builds for windows, but it would be necessary to have this for downstream packages like libsndfile to function comparably across platforms.

**Comment by bmcfee** [comment]
I did a bit of digging into this, and it looks like the recommended build process from the mpg123 devs is to use mingw/msys as the compiler / environment.  This seems to be ...  unusual ... in conda-forge packages (outside of the R universe I guess).

Alternatively, the mpg123 source does provide a CMake port that might be easier to get off the ground, but it might also be out of sync with the main line of mpg123 development.  This of course runs the risk of having two entirely different build processes for different platforms.

Not being a windows dev, I have no real insight or preferences here.  Just documenting what I'm seeing for whoever ends up implementing this.

**Comment by bmcfee** [comment]
Spent a little time fiddling with the cmake port - it looks like it works at least for the libraries (libmpg123 etc) but fails for the command-line app.  This is actually okay for my specific needs (libsndfile linkage), but is probably not great in general.  If someone who actually knows how to work windows build systems could take a look, it'd be very helpful!

**Comment by bmcfee** [comment]
One more follow-up on this for today: I crawled the conda-forge repo, and it appears that the only packages currently depending on mpg123 are `gst-plugins-good` and `yaafe`.

yaafe currently does not ship on windows, but it only depends on libmpg123 (not the cmdline app) according to the [README](https://github.com/Yaafe/Yaafe#from-sources.)

gst-plugins also only depends on libmpg123, but currently skips this on the windows build: https://github.com/conda-forge/gstreamer-feedstock/blob/d4834ff7659f0488cdf0207345ceb12f7a976606/recipe/meta.yaml#L195

TLDR: I see no immediate downsides to only shipping the library build and not the application, aside from the general bad look for disparate functionality across platforms.  Adding even this much would remove a barrier to shipping windows packages on all dependents.

For funsies, I'll tag @scopatz for input on this, since his name is popping up in a bunch of these repos.

**Comment by hmaarrfk** [comment]
Chiming in, using msys2 will make it really hard to integrate with other packages due to ABI incompatibility.

I would suggest looking at what we did with ffmpeg as an example of a package with no upstream recommended of using visual studio compatible tools.

**Comment by bmcfee** [comment]
Thanks @hmaarrfk - that was my guess as well from reading the conda-forge dev docs, but I'm pretty far out of my element with this stuff.

That said, I did take a stab at a PR #30 using the cmake port (builds with VS2019), and as far as I can tell, it looks like it worked.

**Comment by hmaarrfk** [comment]
Great job!

**Comment by hmaarrfk** [comment]
yo ushould add yourself as a maintainer, and wait 1 week, then you can ping core if nobody merges.

**Comment by bmcfee** [comment]
Fixed by #30
